diff --git a/rules.module b/rules.module
index 719852c..377ed35 100644
--- a/rules.module
+++ b/rules.module
@@ -4,6 +4,11 @@
  * @file Rules engine module
  */
 
+// The class autoloader may fail for classes added in 7.x-2.4 (Issue 2090511).
+if (!drupal_autoload_class('RulesEventHandlerEntityBundle')) {
+  require_once dirname(__FILE__) . '/includes/rules.event.inc';
+}
+
 // Include our hook implementations early, as they can be called even before
 // hook_init().
 require_once dirname(__FILE__) . '/modules/events.inc';
